### PR TITLE
Issue 4823: Pravega 0.8 does not build on JDK14 - Upgrade Lombok to 1.18.12

### LINKED ISSUE #4823: Pravega 0.8 does not build on JDK14 - Lombok issues
**Problem description**
You cannot build current master of Pravega with JDK14, due to a compatibility issue with Lombok

**Problem location**
Current version of Lombok does not work with JDK14

**Suggestions for an improvement**
Upgrade to latest version of Lombok

### PR DESCRIPTION
**Pull request by eolivelli**
**Change log description**  
- Upgrade Lombok from 1.18.4 to 1.18.12
- adapt code to changes to Lombok (Wither annodation and Builder.Default annotation)

**Purpose of the change**  
Fixes #4823

**What the code does**  
Upgrade Lombok
Deal with Withen annotation renamed to "With"
Deal with new Builder.Default behaviour

**How to verify it**  
Tests should pass at least on JDK8

### CODE REVIEW

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 1]
```diff
@@ -127,39 +127,75 @@
...
     private String jksKeyFile;
 
-    public static final class InProcPravegaClusterBuilder {
-        public InProcPravegaCluster build() {
-            //Check for valid combinations of flags
-            //For ZK
-            Preconditions.checkState(isInProcZK || zkUrl != null, "ZkUrl must be specified");
-
-            //For controller
-            Preconditions.checkState(isInProcController || controllerURI != null,
```
Comment (eolivelli) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:137]: fields with a Default value now works differently and you do not have a field that holds the value

**Comment by shrids** [comment]
Thanks for the PR @eolivelli 👍 
Yes, `@Builder.Default` and a custom build() method with validation do not go well together. `EventWriterConfig` solves this with a pattern where only the fields with default values are mentioned inside the `build()` method. Could we use this pattern? 
ref: https://github.com/pravega/pravega/blob/cd8d3e258b28d2161b9f3af8d3b5310484bf5734/client/src/main/java/io/pravega/client/stream/EventWriterConfig.java#L59-L69

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 2]
```diff
@@ -128,6 +118,11 @@
     private String jksKeyFile;
 
     public static final class InProcPravegaClusterBuilder {
+
+        // default values
+        private int containerCount = 4;
```
Comment (eolivelli) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:123]: nice trick @shrids ! 

addressed your comments
thank you

**Comment by shrids** [comment]
test-pravega-continuous-1/740/ is successful.

**Comment by eolivelli** [comment]
@shiveshr regarding the coverage check I don't know what to do, this patch is trivial and the code that I changed was already covered.

**Comment by shrids** [comment]
I feel this PR can be merged. @fpj @tkaitchuck could you take a look at it and take a call?
(Note: I do not have rights to merge if any of the checks is not green)

**Comment by fpj** [comment]
Is the coverage low because of the lombok annotation changes?

**Comment by andreipaduroiu** [comment]
@fpj  Yes. There is no other code touched in this PR.

**Pull request merged**